docs/getting-started-guides/kubeadm.md

```diff
@@ -54,358 +65,556 @@ If you are not constrained, there are some other tools built to give you complet
+[csr] Received signed certificate from the API server, generating KubeConfig...
+[kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/kubelet.conf"
+
+Node join complete:
+* Certificate signing request sent to master and response
+  received.
+* Kubelet informed of new secure connection details.
+
+Run 'kubectl get nodes' on the master to see this machine join.
 ```
 
-The three statuses ```RunContainerError``` and ```CrashLoopBackOff``` and ```ContainerCreating``` are very common. 
+A few seconds later, you should notice this node in the output from `kubectl get
+nodes` when run on the master.
 
-To help diagnose what happened, you can use the following command to check what is in the logs:
+### (Optional) Controlling your cluster from machines other than the master
 
-```bash
-kubectl describe -n kube-system po {YOUR_POD_NAME}
+In order to get a kubectl on some other computer (e.g. laptop) to talk to your
+cluster, you need to copy the kubeconfig file from your master to your
+workstation like this:
+
+``` bash
+scp root@<master ip>:/etc/kubernetes/admin.conf .
+kubectl --kubeconfig ./admin.conf get nodes
 ```
 
-Do not using kubectl logs. You will got the following error:
+**Note:** By default GCE instances disable ssh access for root.  First log in to
+the machine, copy the file someplace that can be accessed and then use [`gcloud
+compute
+copy-files`](https://cloud.google.com/sdk/gcloud/reference/compute/copy-files)
+
+### (Optional) Connecting to the API Server
+
+If you want to connect to the API Server for viewing the dashboard (note: the
+dashboard isn't deployed by default) from outside the cluster you can use
+`kubectl proxy`:
 
+``` bash
+scp root@<master ip>:/etc/kubernetes/admin.conf .
+kubectl --kubeconfig ./admin.conf proxy
 ```
-# kubectl logs -n kube-system canal-node-f0lqp
-Error from server (BadRequest): the server rejected our request for an unknown reason (get pods canal-node-f0lqp)
+
+You can now access the API Server locally at `http://localhost:8001/api/v1`
+
+### (Optional) Installing a sample application
+
+Now it is time to take your new cluster for a test drive.  Sock Shop is a sample
+microservices application that shows how to run and connect a set of services on
+Kubernetes. To learn more about the sample microservices app, see the [GitHub
+README](https://github.com/microservices-demo/microservices-demo).
+
+Note that the Sock Shop demo only works on `amd64`.
+
+``` bash
+kubectl create namespace sock-shop
+kubectl apply -n sock-shop -f "https://github.com/microservices-demo/microservices-demo/blob/master/deploy/kubernetes/complete-demo.yaml?raw=true"
 ```
 
-The ```kubectl describe``` comand gives you more details about the logs
+You can then find out the port that the [NodePort feature of
+services](/docs/user-guide/services/) allocated for the front-end service by
+running:
 
+``` bash
+kubectl describe svc front-end -n sock-shop
```

Comment:
BTW -- sock shop demo outputs errors like:

```
the namespace from the provided object "zipkin" does not match the namespace "sock-shop". You must pass '--namespace=zipkin' to perform this operation.
```

Might want to pass that on to weave folks.